### PR TITLE
docs: draft 4.0.0-beta.1 news

### PR DESCRIPTION
**Pull request by ckerr**
Reviewers, the draft notes can be viewed at https://github.com/transmission/transmission/blob/refactor/update-news/news/news-4.0.0-beta-1.md.

Yes I read through ~1300 commits to make these notes. :dizzy:

### CODE REVIEW

File: news/news-4.0.0-beta-1.md [hunk 1]
```diff
@@ -0,0 +1,464 @@
...
+#### `macOS Client` code contributions:
+
+- [alimony](https://github.com/alimony) ((Markus Amalthea Magnuson):
+  - Added ability to filter on error status. ([#19](https://github.com/transmission/transmission/pull/19))
+- [azy5030](https://github.com/azy5030) (Ali):
+  - Add ability to change piece size during torrent creation ([#2416](https://github.com/transmission/transmission/pull/2416))
+- [Coeur](https://github.com/Coeur) (A Cœur):
+  - macOS client icon improvements ([#3250](https://github.com/transmission/transmission/pull/3250), [#3224](https://github.com/transmission/transmission/pull/3224), [#3094](https://github.com/transmission/transmission/pull/3094), [#3065](https://github.com/transmission/transmission/pull/3065))
+  - Fix QuickLook ([#3001](https://github.com/transmission/transmission/pull/3001))
+  - Support pasting multimple magnet links ([#3087](https://github.com/transmission/transmission/pull/3087), [#3086](https://github.com/transmission/transmission/pull/3086))
```
Comment (Coeur) [inline news/news-4.0.0-beta-1.md:266]: multimple > multiple

File: news/news-4.0.0-beta-1.md [hunk 2]
```diff
@@ -0,0 +1,464 @@
...
+**TODO(ckerr): benchmarks**
+
+RPC payloads are now compressed using [libdeflate](https://github.com/ebiggers/libdeflate), a "heavily optimized [library that is] significantly faster than the zlib library."
+
+### Code modernization
+
+- The entire codebase has been migrated from C to C++. In the process, we were able to remove thousands of lines of custom code and use standard C++ libraries instead. libtransmission's source code size has shrunk by 15% compared to Transmission 3.00.
+- The GTK client has been ported to [gtkmm](http://www.gtkmm.org/en/).
+- The Web client no longer uses jQuery and uses modern JavaScript.
+- The unit tests have been ported to [Google Test](http://bittorrent.org/beps/bep_0052.html). Many new tests, including fuzz testing, have been added. CI has been improved to run sanitizer builds of the tests for new code.
```
Comment (tearfur) [inline news/news-4.0.0-beta-1.md:41]: Wrong link for Google Test?

File: news/news-4.0.0-beta-1.md [hunk 3]
```diff
@@ -0,0 +1,464 @@
...
+* Updated code to use modern JavaScript.
+* No longer use jQuery UI.
+* No longer use jQuery.
+* Use Webpack to bundle the Javascript, CSS, and assets together -- the entire bundle size is now 68K gzipped.
+* Added eslint / prettier / stylelint tooling.
+* Uses torrent-get's 'table' mode for more efficient RPC calls.
+
+### BitTorrent 2.0 File Support
+
+- Transmission 4.0.0 now supports .torrent files that were created with the [BitTorrent v2 spec](http://bittorrent.org/beps/bep_0052.html). 4.0.0 users will be able to download and seed torrents created by both BitTorrent v1 and v2. Support for _creating_ v2 torrents is slated forr the next major release of Transmission.
```
Comment (tearfur) [inline news/news-4.0.0-beta-1.md:63]:
```suggestion
- Transmission 4.0.0 now supports .torrent files that were created with the [BitTorrent v2 spec](http://bittorrent.org/beps/bep_0052.html). 4.0.0 users will be able to download and seed torrents created by both BitTorrent v1 and v2. Support for _creating_ v2 torrents is slated for the next major release of Transmission.
```

**Review by tearfur — CHANGES_REQUESTED** [review]
Very excited about this release. Just want to help out a tiny bit as a user.

File: news/news-4.0.0-beta-1.md (hunk 1 above)
Comment (Coeur) [reply news/news-4.0.0-beta-1.md:266]:
```suggestion
  - Support pasting multiple magnet links ([#3087](https://github.com/transmission/transmission/pull/3087), [#3086](https://github.com/transmission/transmission/pull/3086))
```

Comment (Coeur) [reply news/news-4.0.0-beta-1.md:266]: @ckerr not fixing this one?

**Comment by ckerr** [comment]
> @ckerr what's the mystery deal for the one I commented above? musical_note smile

I don't understand the question?

**Comment by ckerr** [comment]
Even though this is incomplete I think I'm just going to merge it anyway -- that way, other people can contribute PRs to add to it & improve it. There's still probably 2 weeks of coding left before we start freezing for release

**Comment by GaryElshaw** [comment]
This is causing some breaks during CI.

```
CMake Error at cmake_install.cmake:50 (file):
  file INSTALL cannot find "/__w/transmission/transmission/src/NEWS.md": No
  such file or directory.
```

**Pull request merged**